tastypie/fields.py

```diff
@@ -763,6 +764,15 @@ def hydrate(self, bundle):
         if value is None:
             return value
 
+        if bundle.obj and isinstance(getattr(bundle.obj.__class__, self.attribute), SingleRelatedObjectDescriptor):
```

Comment:
If `self.attribute` is a composite attribute (`country__president`), then we get exception here, because our class doesn't have the attribute `country__president`, but have the attribute `country`, which have the attribute `president`). And it's taking into account in `self._attrs`:
```
self._attrs = attribute.split('__') if attribute is not None and isinstance(attribute, six.string_types) else []
```
Maybe I should open issue + test on this case?